### PR TITLE
Add subscription triggers [PREMIUM-246]

### PR DESCRIPTION
**Pull request by websupporter**
## Description

See https://github.com/mailpoet/mailpoet-premium/pull/816

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/816

## Linked tickets

[PREMIUM-246]

## merge notes

Merge after https://github.com/mailpoet/mailpoet/pull/5259

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[PREMIUM-246]: https://mailpoet.atlassian.net/browse/PREMIUM-246?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-renewed/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  __('woocommerce', 'mailpoet'),
+  __('subscription', 'mailpoet'),
+  __('renewed', 'mailpoet'),
+];
+export const step: StepType = {
+  key: 'woocommerce-subscriptions:subscription-renewed',
+  group: 'triggers',
+  title: () => __('Woo Subscription renewed', 'mailpoet'),
+  description: () =>
+    __('Start the automation when a subscription renewed.', 'mailpoet'),
```
Comment (johnolek) [inline mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-renewed/index.tsx:16]: I would prefer `when a user renews their subscription`, or `when a subscription renews` or `when a subscription is renewed`

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-ended/index.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  __('subscription', 'mailpoet'),
+  __('trial', 'mailpoet'),
+  __('ended', 'mailpoet'),
+];
+export const step: StepType = {
+  key: 'woocommerce-subscriptions:trial-ended',
+  group: 'triggers',
+  title: () => __('Woo Subscription trial ended', 'mailpoet'),
+  description: () =>
+    __('Start the automation when a subscription trial ended.', 'mailpoet'),
```
Comment (johnolek) [inline mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-ended/index.tsx]: `when a subscription trial ends` sounds better to me

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-ended/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  icon: () => <Icon />,
+  edit: () => (
+    <PremiumModalForStepEdit
+      tracking={{
+        utm_medium: 'upsell_modal',
+        utm_campaign: 'create_automation_editor_woo_subscription_trial_ended',
+      }}
+    >
+      {__(
+        'Starting an automation when a subscription trial ended is a premium feature.',
```
Comment (johnolek) [inline mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-ended/index.tsx:32]:
```suggestion
        'Starting an automation when a subscription trial ends is a premium feature.',
```

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-started/index.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  __('subscription', 'mailpoet'),
+  __('trial', 'mailpoet'),
+  __('started', 'mailpoet'),
+];
+export const step: StepType = {
+  key: 'woocommerce-subscriptions:trial-started',
+  group: 'triggers',
+  title: () => __('Woo Subscription trial started', 'mailpoet'),
+  description: () =>
+    __('Start the automation when a subscription trial started.', 'mailpoet'),
```
Comment (johnolek) [inline mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-started/index.tsx]:
```suggestion
    __('Start the automation when a subscription trial starts.', 'mailpoet'),
```

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-started/index.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  icon: () => <Icon />,
+  edit: () => (
+    <PremiumModalForStepEdit
+      tracking={{
+        utm_medium: 'upsell_modal',
+        utm_campaign: 'create_automation_editor_woo_subscription_trial_ended',
+      }}
+    >
+      {__(
+        'Starting an automation when a subscription trial started is a premium feature.',
```
Comment (johnolek) [inline mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-started/index.tsx]:
```suggestion
        'Starting an automation when a subscription trial starts is a premium feature.',
```

**Review by johnolek — CHANGES_REQUESTED** [review]
Just a few phrasing suggestions

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-renewed/index.tsx (hunk 1 above)
Comment (websupporter) [reply mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-renewed/index.tsx:16]: https://github.com/mailpoet/mailpoet/pull/5264/commits/d24111eda686d0e5643dd27b42db2c359aac3849

File: mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-ended/index.tsx (hunk 2 above)
Comment (websupporter) [reply mailpoet/assets/js/src/automation/integrations/woocommerce-subscriptions/steps/subscription-trial-ended/index.tsx]: https://github.com/mailpoet/mailpoet/pull/5264/commits/d24111eda686d0e5643dd27b42db2c359aac3849

**Comment by websupporter** [comment]
Thanks @johnolek! Updated the wording.

**Comment by johnolek** [comment]
Looks like there are conflicts, @websupporter

**Comment by websupporter** [comment]
Oh, seems I didn't force push my rebase. Rebased @johnolek

**Review by johnolek — DISMISSED** [review]

**Comment by websupporter** [comment]
rebased

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**